clients/spark/build.sbt

```
@@ -112,26 +112,34 @@ val publishBucket = settingKey[String]("Target S3 bucket for publishing the JAR"
 publishBucket := sys.props.get("publish.bucket").filter(_.nonEmpty).getOrElse("treeverse-clients-us-east")
 
 s3Upload := {
-  import sys.process._
+  import software.amazon.awssdk.core.sync.RequestBody
+  import software.amazon.awssdk.regions.Region
+  import software.amazon.awssdk.services.s3.S3Client
+  import software.amazon.awssdk.services.s3.model.{PutObjectRequest, S3Exception}
 
   val log = streams.value.log
   val bucket = publishBucket.value
   val jarFile = (assembly / assemblyOutputPath).value
   val key = s"${name.value}/${version.value}/${(assembly / assemblyJarName).value}"
 
-  val cmd = Seq(
-    "aws","s3api","put-object",
-    "--bucket", bucket,
-    "--key", key,
-    "--body", jarFile.getAbsolutePath,
-    "--if-none-match","*",
-    "--region", "us-east-1",
-    "--acl","public-read" // TODO: remove after switching bucket to "Bucket owner enforced"
-  )
-
-  val pl = ProcessLogger(out => log.info(out), err => log.error(err))
-  Process(cmd).!!(pl)
-  log.info(s"Uploaded to S3 successfully: https://$bucket.s3.amazonaws.com/$key")
+  val s3 = S3Client.builder().region(Region.of("us-east-1")).build()
+  try {
+    val req = PutObjectRequest.builder()
+      .bucket(bucket)
+      .key(key)
+      .ifNoneMatch("*")
+      .build()
+
+    s3.putObject(req, RequestBody.fromFile(jarFile))
+    log.info(s"Uploaded to S3 successfully: s3://$bucket/$key")
+  } catch {
+    case e: S3Exception if e.statusCode() == 412 =>
+      sys.error(s"Artifact already exists: s3://$bucket/$key")
```

Great error message!  But it loses the original exception, which might contain more details.  I _strongly_ recommend including it as the cause of the RuntimeException:
```suggestion
      throw new RuntimeException(s"Artifact already exists: s3://$bucket/$key", e)
```